packages/core/src/index.ts

```diff
@@ -602,8 +611,10 @@ class Hopr extends EventEmitter {
    * Gets the observed addresses of a given peer.
    * @param peer peer to query for
    */
-  public getObservedAddresses(peer: PeerId): Multiaddr[] {
-    return (this.libp2p.peerStore.get(peer)?.addresses ?? []).map((addr) => addr.multiaddr)
+  public async getObservedAddresses(peer: PeerId): Promise<Multiaddr[]> {
+    const addresses = await this.libp2p.peerStore.addressBook.get(peer)
+    const multiAddresses = addresses.map((addr) => addr.multiaddr)
```

Comment:
Just return this without creating a variable.

Reply:
Not much of a fan of calling functions in the return statements.

Reply:
Adapted.